### DISCUSSION TITLE
嵌套effect的问题

### DISCUSSION THREAD

**Discussion by atlasyoung** [cuiyiming1998/Vue3-mini#5] [Q&A] [answered]
如果effect存在嵌套的情况, 内层的effect会覆盖 `activeEffect`, 这种情况如何解决呢?

**Comment by cuiyiming1998** [comment, marked as answer]
在effect里还有一个effectStack栈 每次执行`run`的时候会push进去, activeEffect执行完毕后会pop出栈

始终让active指向栈顶的元素, 这样就可以避免嵌套时覆盖的问题了
